### PR TITLE
Rek convert s3 script

### PR DESCRIPTION
**Pull request by kilbergr**
This PR is for issue #2139 . The PR results in creating a series of files corresponding to this directory structure: https://docs.google.com/document/d/1SgaPo6eQvJzDmkl790jM4sZ_NAsB4qaySRVolRCfwsU/edit#

The fabfile changes are mostly the result of having run through with flake8/black.

If this looks like too much to go through, I'm happy to do a code walk thru or reassign to Michael. We thought it might be better to get some fresh eyes on it.

To run the script, run `fab export_cases_to_s3` from the CAP docker web machine. This is set to work exclusively on Reporter 528 currently, but will ultimately go through each reporter in chunks.

### CODE REVIEW

File: capstone/scripts/convert_s3.py [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+from collections import namedtuple
+
+from capapi.documents import CaseDocument
+from capapi.serializers import NoLoginCaseDocumentSerializer, CaseDocumentSerializer
+from capdb.models import Reporter, VolumeMetadata, CaseMetadata
+
+s3_client = boto3.client("s3")
+
+
+# Doing this with the API because there are many fewer than cases
```
Comment (matteocargnelutti) [inline capstone/scripts/convert_s3.py]: Very minor thing: This function would benefit from a docstring `""""` intro comment, like you did for the other functions in this module :).

File: capstone/scripts/convert_s3.py [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+def put_volumes_reporters_on_s3(redacted):
+    bucket = get_bucket_name(redacted)
+
+    for file_type in ["reporters", "volumes"]:
+        current_endpoint = f"https://api.case.law/v1/{file_type}/"
```
Comment (matteocargnelutti) [inline capstone/scripts/convert_s3.py]: Another minor thing, which could help with testing:
There are multiple functions in this file in which `https://api.case.law/v1/` is hardcoded. 
Hoisting this into a variable which can be replaced at runtime could help with testing. For example, you could decide to make it point to a different host (local?) which hosts mock files.

File: capstone/scripts/convert_s3.py [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+            Bucket=bucket,
+            Key=key,
+            ContentType=content_type,
+            ChecksumSHA256=sha256_hash,
+        )
+    except ClientError as err:
+        raise Exception(f"Error uploading {key}: %s" % err)
+
+
+def get_bucket_name(redacted):
```
Comment (matteocargnelutti) [inline capstone/scripts/convert_s3.py]: I think type annotations could give a clarity boost to this script. 
For example: 
```python
def get_bucket_name(redacted: bool) -> str:
```

Tells me a lot more about what this function does at a glance.

**Review by matteocargnelutti — APPROVED** [review]
@kilbergr This is very cool 👏 !

This looks good to me, but I don't think I am knowledgable enough about both CAP and your current goals to give you more than "the code looks good to me"-level feedback. I made some suggestions in that spirit. 

Approving this PR, but of course feel free to tag someone else for another round of review 😃

**Comment by kilbergr** [comment]
I'm waiting for @jcushman's ok after looking at the generated volume/reporter packages.

**Comment by kilbergr** [comment]
Closing this PR in favor of a new PR to an experimental branch: https://github.com/harvard-lil/capstone/pull/2150
That way, we can actually merge it.

**Pull request closed without merging**